core/storage/exploration/gae_models.py

```diff
@@ -1113,3 +1135,121 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+    """Version history model for an oppia exploration.
+
+    Version history means some information about the previous commit on each
+    state and the exploration metadata at a particular version of the
+    exploration. The information about each state comprises of the version
+    number of the exploration on which the state was previously edited,
+    the name of the state at the previous version and the id of the user who
+    committed those changes. For metadata, the information comprises of the
+    version number of the explortion on which the metadata was previously
+    edited and the id of the user who committed those changes.
+
+    A new instance of this model is created each time a new exploration
+    is created or some changes are saved in an exploration.
+
+    The id of the model is generated as follows:
+    {exploration_id}-{exploration_version}
+    """
+
+    # The id of the corresponding exploration.
+    exploration_id = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The version of the corresponding exploration.
+    exploration_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The details of the previous commit on each state at a particular
+    # version of the exploration. The json structure will look like the
+    # following:
+    # {
+    #   [state_name: str]: {
+    #     "previously_edited_in_version": int,
+    #     "state_name_in_previous_version": str,
+    #     "committer_id": str
+    #   }
+    # }
+    # The json object can have multiple keys in this case depending on
+    # the number of states.
+    state_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The details of the previous commit on the exploration metadata at a
+    # particular version of the exploration. The json structure will look
+    # like the following:
+    # {
+    #   "previously_edited_in_version": str,
+    #   "committer_id": str.
+    # }
+    # The json object will have only two keys in this case.
+    metadata_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The user ids of the users who did the 'previous commit' on each state
+    # in a particular version of the exploration. It is required during the
+    # wipeout process to query for the models efficiently.
+    committer_ids = datastore_services.StringProperty(
+        indexed=True, repeated=True)
+
+    @classmethod
+    def get_instance_id(cls, exp_id: str, exp_version: int) -> str:
+        """Returns ID of the exploration version history model.
+
+        Args:
+            exp_id: str. The ID of the exploration.
+            exp_version: int. The version of the exploration.
+
+        Returns:
+            str. A string containing exploration ID and
+            exploration version.
+        """
+        return 'version-history-%s-%s' % (exp_id, exp_version)
```

Comment:
Do you actually need the "version-history-" prefix?

(I think I don't mind it, but just wondering.)

Also some old explorations have "-" and "_" in their ID so you might want to use a different delimiter, like "."

Reply:
I have added the prefix so that the ids are distinguished from other models such as the SnapshotMetadata and SnapshotContent models.